purchase_order_no_zero_price/models/purchase_order_line.py

```diff
@@ -0,0 +1,29 @@
+# Copyright (C) 2022 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, exceptions, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = "purchase.order.line"
+
+    @api.constrains("price_unit", "display_type", "state")
+    def _check_price_unit_zero(self):
+        # Prevent zero prices on confirmed POs
+        # to help ensure proper Stock Valuation
+        # when receiving the products
+        # Also supports `purchase_triple_discount` module
+        no_price_lines = self.filtered(
```

Comment:
@HviorForgeFlow @dsolanki-initos IMHO this should be configurable either on the company level, either on product one as it breaks standard behavior.

Modules that are not aware of this one (no dependency) could fail.